scipy/interpolate/_bsplines.py

```diff
@@ -33,6 +34,44 @@ def _as_float_array(x, check_finite=False):
         raise ValueError("Array must not contain infs or nans.")
     return x
 
+def _find_left(t, x, k):
```

Comment:
This whole function can likely be ditched by moving the source to cython and using `find_interval` which is available in `_bspl.pyx`. A similar thing was done in gh-14344